### ISSUE TITLE
 needless_doctest_main reports wrong line if there are imports

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5236]
````rust
/// # Examples
///
/// ```
/// use std::path::PathBuf;
/// use std::vec;
///
/// fn main() {
/// }
///
/// ```

fn main() {
    println!("Hello, world!");
}

````

It looks like the lint assumes the `fn main()` will always be in the first line of the code block..?

````
warning: needless `fn main` in doctest
 --> src/main.rs:4:4
  |
4 | /// use std::path::PathBuf;
  |    ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::needless_doctest_main)]` on by default
````

This causes strange looking warnings.

`clippy 0.0.212 (fc5d0cc 2020-02-24)`

**Comment by llogiq** [comment]
I think I know the solution for this...let me code it up, just a sec.

**Comment by llogiq** [comment]
(fix was wrongly marked closed by github)
